subsys/net/ip/route.c

```diff
@@ -1079,13 +1079,15 @@ int net_route_packet(struct net_pkt *pkt, struct in6_addr *nexthop)
 	 * destination address to be the nexthop recipient.
 	 */
 	if (is_ll_addr_supported(net_pkt_iface(pkt))) {
-		net_pkt_lladdr_src(pkt)->addr = net_pkt_lladdr_if(pkt)->addr;
```

Comment:
Maybe just memcpy entire struct, instead of copying each field separately? Same below.